src/include/daos_api.h

```diff
@@ -16,6 +16,8 @@
 extern "C" {
 #endif
 
+#include <daos_types.h>
+
```

Comment:
so even though those headers are installed, i believe we discussed at one point that really the user should just include daos.h for the daos API, then add daos_fs.h, or others for the higher level APIs vs being able to add each header. so your changes to make each public header byitself pull the other needed ones is not necessary.